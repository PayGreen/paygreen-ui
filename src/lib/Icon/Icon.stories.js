import React from 'react';
import {
    PlaneIcon,
    OrganizationIcon,
    ShopIcon,
    TreesIcon,
    PointerIcon
} from './Icon';
import {
    colorPalletOptions,
    colorPalletDefault,
    colorThemeOptions,
    colorThemeDefault,
    greyOptions,
    greyDefault,
    formStatusOptions,
    formStatusDefault,
    iconSizeOptions,
    spaceOptions
} from '../../shared/constants';
import { storiesOf } from '@storybook/react';
import { withKnobs, boolean, select, radios } from '@storybook/addon-knobs';

const colorPalletLabel = 'Color pallet';
const colorThemeLabel = 'Color theme';
const colorWabLabel = 'Grey color';
const colorStatusLabel = 'Status color';
const iconSizeLabel = 'Icon size';
const backgroundLabel = 'Has background';
const shadowLabel = 'With shadow';
const activeLabel = 'Active';
const marginTopLabel = 'Margin top';
const marginBottomLabel = 'Margin bottom';
const marginLeftLabel = 'Margin left';
const marginRightLabel = 'Margin right';
const centeredLabel = 'Centered';

storiesOf('Icons', module)
    .addDecorator(withKnobs)
    .add('Some icons', () => (
        <div>
            <PlaneIcon
                colorPallet={radios(colorPalletLabel, colorPalletOptions, colorPalletDefault)}
                colorTheme={select(colorThemeLabel, colorThemeOptions, colorThemeDefault)}
                colorWab={select(colorWabLabel, greyOptions, greyDefault)}
                colorStatus={select(colorStatusLabel, formStatusOptions, formStatusDefault)}
                iconSize={select(iconSizeLabel, iconSizeOptions, iconSizeOptions.lg)}
                hasBackground={boolean(backgroundLabel, false)}
                shadowSize={boolean(shadowLabel, false)}
                isActive={boolean(activeLabel, false)}
                isCentered={boolean(centeredLabel, false)}
                marginTop={select(marginTopLabel, spaceOptions, spaceOptions.sm)}
                marginBottom={select(marginBottomLabel, spaceOptions, spaceOptions.sm)}
                marginLeft={select(marginLeftLabel, spaceOptions, spaceOptions.sm)}
                marginRight={select(marginRightLabel, spaceOptions, spaceOptions.sm)}
            />

            <OrganizationIcon
                colorPallet={radios(colorPalletLabel, colorPalletOptions, colorPalletDefault)}
                colorTheme={select(colorThemeLabel, colorThemeOptions, colorThemeDefault)}
                colorWab={select(colorWabLabel, greyOptions, greyDefault)}
                colorStatus={select(colorStatusLabel, formStatusOptions, formStatusDefault)}
                iconSize={select(iconSizeLabel, iconSizeOptions, iconSizeOptions.lg)}
                hasBackground={boolean(backgroundLabel, false)}
                shadowSize={boolean(shadowLabel, false)}
                isActive={boolean(activeLabel, false)}
                isCentered={boolean(centeredLabel, false)}
                marginTop={select(marginTopLabel, spaceOptions, spaceOptions.sm)}
                marginBottom={select(marginBottomLabel, spaceOptions, spaceOptions.sm)}
                marginLeft={select(marginLeftLabel, spaceOptions, spaceOptions.sm)}
                marginRight={select(marginRightLabel, spaceOptions, spaceOptions.sm)}
            />

            <ShopIcon
                colorPallet={radios(colorPalletLabel, colorPalletOptions, colorPalletDefault)}
                colorTheme={select(colorThemeLabel, colorThemeOptions, colorThemeDefault)}
                colorWab={select(colorWabLabel, greyOptions, greyDefault)}
                colorStatus={select(colorStatusLabel, formStatusOptions, formStatusDefault)}
                iconSize={select(iconSizeLabel, iconSizeOptions, iconSizeOptions.lg)}
                hasBackground={boolean(backgroundLabel, false)}
                shadowSize={boolean(shadowLabel, false)}
                isActive={boolean(activeLabel, false)}
                isCentered={boolean(centeredLabel, false)}
                marginTop={select(marginTopLabel, spaceOptions, spaceOptions.sm)}
                marginBottom={select(marginBottomLabel, spaceOptions, spaceOptions.sm)}
                marginLeft={select(marginLeftLabel, spaceOptions, spaceOptions.sm)}
                marginRight={select(marginRightLabel, spaceOptions, spaceOptions.sm)}
            />

            <TreesIcon
                colorPallet={radios(colorPalletLabel, colorPalletOptions, colorPalletDefault)}
                colorTheme={select(colorThemeLabel, colorThemeOptions, colorThemeDefault)}
                colorWab={select(colorWabLabel, greyOptions, greyDefault)}
                colorStatus={select(colorStatusLabel, formStatusOptions, formStatusDefault)}
                iconSize={select(iconSizeLabel, iconSizeOptions, iconSizeOptions.lg)}
                hasBackground={boolean(backgroundLabel, false)}
                shadowSize={boolean(shadowLabel, false)}
                isActive={boolean(activeLabel, false)}
                isCentered={boolean(centeredLabel, false)}
                marginTop={select(marginTopLabel, spaceOptions, spaceOptions.sm)}
                marginBottom={select(marginBottomLabel, spaceOptions, spaceOptions.sm)}
                marginLeft={select(marginLeftLabel, spaceOptions, spaceOptions.sm)}
                marginRight={select(marginRightLabel, spaceOptions, spaceOptions.sm)}
            />

            <PointerIcon
                colorPallet={radios(colorPalletLabel, colorPalletOptions, colorPalletDefault)}
                colorTheme={select(colorThemeLabel, colorThemeOptions, colorThemeDefault)}
                colorWab={select(colorWabLabel, greyOptions, greyDefault)}
                colorStatus={select(colorStatusLabel, formStatusOptions, formStatusDefault)}
                iconSize={select(iconSizeLabel, iconSizeOptions, iconSizeOptions.lg)}
                hasBackground={boolean(backgroundLabel, false)}
                shadowSize={boolean(shadowLabel, false)}
                isActive={boolean(activeLabel, false)}
                isCentered={boolean(centeredLabel, false)}
                marginTop={select(marginTopLabel, spaceOptions, spaceOptions.sm)}
                marginBottom={select(marginBottomLabel, spaceOptions, spaceOptions.sm)}
                marginLeft={select(marginLeftLabel, spaceOptions, spaceOptions.sm)}
                marginRight={select(marginRightLabel, spaceOptions, spaceOptions.sm)}
            />
        </div>
    ), {
        notes: 'Queen icons with some styles. You can add background, shadow, change colors, add active mode, center icon, change margins or icon size.',
    });