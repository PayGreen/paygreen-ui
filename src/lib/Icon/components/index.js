import ArrowBottom from './ArrowBottom';
import ArrowCircleLeft from './ArrowCircleLeft';
import ArrowCircleLeftBold from './ArrowCircleLeftBold';
import ArrowCircleRightBold from './ArrowCircleRightBold';
import ArrowCircleRight from './ArrowCircleRight';
import ArrowLeft from './ArrowLeft';
import ArrowLongBottom from './ArrowLongBottom';
import ArrowLongLeft from './ArrowLongLeft';
import ArrowLongRight from './ArrowLongRight';
import ArrowLongTop from './ArrowLongTop';
import ArrowRight from './ArrowRight';
import ArrowRoundBottomLeft from './ArrowRoundBottomLeft';
import ArrowRoundBottomRight from './ArrowRoundBottomRight';
import ArrowRoundTopLeft from './ArrowRoundTopLeft';
import ArrowRoundTopRight from './ArrowRoundTopRight';
import ArrowTop from './ArrowTop';
import BalloonBold from './BalloonBold';
import Balloon from './Balloon';
import Bank from './Bank';
import BeakerBold from './BeakerBold';
import Beaker from './Beaker';
import BellBold from './BellBold';
import Bell from './Bell';
import BulbBold from './BulbBold';
import Bulb from './Bulb';
import CardsBold from './CardsBold';
import Cards from './Cards';
import Car from './Car';
import Cart from './Cart';
import CautionBold from './CautionBold';
import Caution from './Caution';
import Change from './Change';
import CheckBold from './CheckBold';
import Checkbox from './Checkbox';
import Check from './Check';
import ClockBold from './ClockBold';
import Clock from './Clock';
import Code from './Code';
import Coins from './Coins';
import Contract from './Contract';
import CrossBold from './CrossBold';
import Cross from './Cross';
import Download from './Download';
import Empty from './Empty';
import Eye from './Eye';
import Factory from './Factory';
import Globe from './Globe';
import Heart from './Heart';
import Hourglass from './Hourglass';
import Information from './Information';
import Leaf from './Leaf';
import LetterBold from './LetterBold';
import Letter from './Letter';
import LinkBold from './LinkBold';
import Link from './Link';
import List from './List';
import Lock from './Lock';
import Marker from './Marker';
import Meal from './Meal';
import Megaphone from './Megaphone';
import MenuBold from './MenuBold';
import MenuDots from './MenuDots';
import Menu from './Menu';
import Meter from './Meter';
import Minus from './Minus';
import Moon from './Moon';
import Moove from './Moove';
import Organization from './Organization';
import Out from './Out';
import Paperclip from './Paperclip';
import Pen from './Pen';
import Phone from './Phone';
import Picture from './Picture';
import Plane from './Plane';
import Plus from './Plus';
import Pointer from './Pointer';
import Popin from './Popin';
import QuestionBold from './QuestionBold';
import Question from './Question';
import QuoteLeft from './QuoteLeft';
import QuoteRight from './QuoteRight';
import RefreshBold from './RefreshBold';
import Refresh from './Refresh';
import Satchel from './Satchel';
import Schedule from './Schedule';
import SearchBold from './SearchBold';
import Search from './Search';
import Shop from './Shop';
import Sun from './Sun';
import Support from './Support';
import Trees from './Trees';
import Trophy from './Trophy';
import Truck from './Truck';
import Tv from './Tv';
import UserBold from './UserBold';
import Users from './Users';
import User from './User';
import Wallet from './Wallet';
import Weight from './Weight';
import Wheels from './Wheels';

export {
    ArrowBottom,
    ArrowCircleLeft,
    ArrowCircleLeftBold,
    ArrowCircleRightBold,
    ArrowCircleRight,
    ArrowLeft,
    ArrowLongBottom,
    ArrowLongLeft,
    ArrowLongRight,
    ArrowLongTop,
    ArrowRight,
    ArrowRoundBottomLeft,
    ArrowRoundBottomRight,
    ArrowRoundTopLeft,
    ArrowRoundTopRight,
    ArrowTop,
    BalloonBold,
    Balloon,
    Bank,
    BeakerBold,
    Beaker,
    BellBold,
    Bell,
    BulbBold,
    Bulb,
    CardsBold,
    Cards,
    Car,
    Cart,
    CautionBold,
    Caution,
    Change,
    CheckBold,
    Checkbox,
    Check,
    ClockBold,
    Clock,
    Code,
    Coins,
    Contract,
    CrossBold,
    Cross,
    Download,
    Empty,
    Eye,
    Factory,
    Globe,
    Heart,
    Hourglass,
    Information,
    Leaf,
    LetterBold,
    Letter,
    LinkBold,
    Link,
    List,
    Lock,
    Marker,
    Meal,
    Megaphone,
    MenuBold,
    MenuDots,
    Menu,
    Meter,
    Minus,
    Moon,
    Moove,
    Organization,
    Out,
    Paperclip,
    Pen,
    Phone,
    Picture,
    Plane,
    Plus,
    Pointer,
    Popin,
    QuestionBold,
    Question,
    QuoteLeft,
    QuoteRight,
    RefreshBold,
    Refresh,
    Satchel,
    Schedule,
    SearchBold,
    Search,
    Shop,
    Sun,
    Support,
    Trees,
    Trophy,
    Truck,
    Tv,
    UserBold,
    Users,
    User,
    Wallet,
    Weight,
    Wheels,
};
