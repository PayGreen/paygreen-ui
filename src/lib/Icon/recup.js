var fs = require('fs');

const icons = [
    'arrow-bottom.svg',
    'arrow-circle-left-bold.svg',
    'arrow-circle-left.svg',
    'arrow-circle-right-bold.svg',
    'arrow-circle-right.svg',
    'arrow-left.svg',
    'arrow-long-bottom.svg',
    'arrow-long-left.svg',
    'arrow-long-right.svg',
    'arrow-long-top.svg',
    'arrow-right.svg',
    'arrow-round-bottom-left.svg',
    'arrow-round-bottom-right.svg',
    'arrow-round-top-left.svg',
    'arrow-round-top-right.svg',
    'arrow-top.svg',
    'balloon-bold.svg',
    'balloon.svg',
    'bank.svg',
    'beaker-bold.svg',
    'beaker.svg',
    'bell-bold.svg',
    'bell.svg',
    'bulb-bold.svg',
    'bulb.svg',
    'cards-bold.svg',
    'cards.svg',
    'car.svg',
    'cart.svg',
    'caution-bold.svg',
    'caution.svg',
    'check-bold.svg',
    'checkbox.svg',
    'check.svg',
    'clock-bold.svg',
    'clock.svg',
    'code.svg',
    'coins.svg',
    'contract.svg',
    'cross-bold.svg',
    'cross.svg',
    'download.svg',
    'empty.svg',
    'eye.svg',
    'globe.svg',
    'heart.svg',
    'hourglass.svg',
    'information.svg',
    'leaf.svg',
    'letter-bold.svg',
    'letter.svg',
    'link-bold.svg',
    'link.svg',
    'lock.svg',
    'meal.svg',
    'menu-bold.svg',
    'menu.svg',
    'meter.svg',
    'organization.svg',
    'out.svg',
    'paperclip.svg',
    'pen.svg',
    'phone.svg',
    'picture.svg',
    'plane.svg',
    'pointer.svg',
    'popin.svg',
    'question-bold.svg',
    'question.svg',
    'quote-left.svg',
    'quote-right.svg',
    'refresh-bold.svg',
    'refresh.svg',
    'satchel.svg',
    'schedule.svg',
    'search-bold.svg',
    'search.svg',
    'shop.svg',
    'trees.svg',
    'trophy.svg',
    'truck.svg',
    'user-bold.svg',
    'users.svg',
    'user.svg',
    'wallet.svg',
    'wheels.svg']


icons.forEach((value) => {
    console.log('svg/' + value);
    const data = fs.readFileSync('svg/' + value, 'UTF-8');
    fs.appendFile('test.js', data, function (err) {
        if (err) throw err;
        console.log('Updated!');
    });
})
