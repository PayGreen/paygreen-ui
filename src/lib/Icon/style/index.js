import styled from 'styled-components';
import { directionalProperty } from 'polished';
import { svgFill } from './constants';
import {
    backgroundStyle,
    activeStyle
} from './base';

const IconBase = styled.span`
    display: ${props => props.centered ? 'flex' : 'inline-flex'};
    position: relative;
    width: ${props => props.theme.iconSize[props.iconSize]};
    height: ${props => props.theme.iconSize[props.iconSize]};
    ${props => directionalProperty('margin',
        props.theme.space[props.marginTop],
        props.centered ? 'auto' : props.theme.space[props.marginRight],
        props.theme.space[props.marginBottom],
        props.centered ? 'auto' : props.theme.space[props.marginLeft]
    )};
    transition: all ${props => props.theme.transition.sm};

    svg {
        width: 100%;
        fill: ${props => svgFill[props.colorPallet]};
        transition: all ${props => props.theme.transition.sm};
    }

    ${props => props.background ? backgroundStyle : null};
    ${props => props.active ? activeStyle[props.colorPallet] : null};
`;

export { IconBase };