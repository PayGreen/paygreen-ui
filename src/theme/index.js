import { ThemeDefault } from './theme.default';
import { ThemeDark } from './theme.dark';

export { ThemeDefault, ThemeDark };
