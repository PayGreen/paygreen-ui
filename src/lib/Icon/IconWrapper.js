import React from 'react';
import PropTypes from 'prop-types';
import {
    colorPalletOptions,
    colorPalletDefault,
    colorThemeOptions,
    colorThemeDefault,
    greyOptions,
    greyDefault,
    formStatusOptions,
    formStatusDefault,
    iconSizeOptions,
    spaceOptions,
    spaceDefault,
    iconHtmlTagOptions,
    iconHtmlTagDefault,
    rotateSizeOptions,
    rotateSizeDefault,
} from '../../shared/constants';
import { IconBase } from './style';

const maxDisplayed = 99;

const IconWrapper = ({ children, className, htmlTag, ...rest }) => {
    return (
        <IconBase
            {...rest}
            as={htmlTag}
            className={className ? 'icon ' + className : 'icon'}
        >
            {React.Children.map(children, child => {
                if (typeof child == 'object') {
                    return React.cloneElement(child);
                }
            })}

            {rest.number > 0 &&
            ![iconSizeOptions.xs, iconSizeOptions.xxs].includes(
                rest.iconSize,
            ) ? (
                <span className="badge-number">
                    {rest.number <= maxDisplayed
                        ? rest.number
                        : maxDisplayed + '+'}
                </span>
            ) : null}
        </IconBase>
    );
};

IconWrapper.propTypes = {
    htmlTag: PropTypes.oneOf(Object.values(iconHtmlTagOptions)),
    number: PropTypes.number,
    colorPallet: PropTypes.oneOf(Object.values(colorPalletOptions)),
    colorTheme: PropTypes.oneOf(Object.values(colorThemeOptions)),
    colorWab: PropTypes.oneOf(Object.values(greyOptions)),
    colorStatus: PropTypes.oneOf(Object.values(formStatusOptions)),
    iconSize: PropTypes.oneOf(Object.values(iconSizeOptions)),
    hasBackground: PropTypes.bool,
    hasHover: PropTypes.bool,
    isActive: PropTypes.bool,
    isCentered: PropTypes.bool,
    marginTop: PropTypes.oneOf(Object.values(spaceOptions)),
    marginBottom: PropTypes.oneOf(Object.values(spaceOptions)),
    marginLeft: PropTypes.oneOf(Object.values(spaceOptions)),
    marginRight: PropTypes.oneOf(Object.values(spaceOptions)),
    rotateSize: PropTypes.oneOf(Object.values(rotateSizeOptions)),
};

IconWrapper.defaultProps = {
    htmlTag: iconHtmlTagDefault,
    number: 0,
    colorPallet: colorPalletDefault,
    colorTheme: colorThemeDefault,
    colorWab: greyDefault,
    colorStatus: formStatusDefault,
    iconSize: iconSizeOptions.sm,
    hasBackground: false,
    hasHover: false,
    isActive: false,
    isCentered: false,
    marginTop: spaceDefault,
    marginBottom: spaceDefault,
    marginLeft: spaceDefault,
    marginRight: spaceDefault,
    rotateSize: rotateSizeDefault,
};

export default IconWrapper;
