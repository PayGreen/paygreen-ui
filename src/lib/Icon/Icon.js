import React from 'react';
import IconWrapper from './IconWrapper';
import {
  ArrowBottom,
  ArrowCircleLeft,
  ArrowCircleLeftBold,
  ArrowCircleRightBold,
  ArrowCircleRight,
  ArrowLeft,
  ArrowLongBottom,
  ArrowLongLeft,
  ArrowLongRight,
  ArrowLongTop,
  ArrowRight,
  ArrowRoundBottomLeft,
  ArrowRoundBottomRight,
  ArrowRoundTopLeft,
  ArrowRoundTopRight,
  ArrowTop,
  BalloonBold,
  Balloon,
  Bank,
  BeakerBold,
  Beaker,
  BellBold,
  Bell,
  BulbBold,
  Bulb,
  CardsBold,
  Cards,
  Car,
  Cart,
  CautionBold,
  Caution,
  Change,
  CheckBold,
  Checkbox,
  Check,
  ClockBold,
  Clock,
  Code,
  Coins,
  Contract,
  CrossBold,
  Cross,
  Download,
  Empty,
  Eye,
  Factory,
  Globe,
  Heart,
  Hourglass,
  Information,
  Leaf,
  LetterBold,
  Letter,
  LinkBold,
  Link,
  List,
  Lock,
  Marker,
  Meal,
  Megaphone,
  MenuBold,
  MenuDots,
  Menu,
  Meter,
  Minus,
  Moon,
  Moove,
  Organization,
  Out,
  Paperclip,
  Pen,
  Phone,
  Picture,
  Plane,
  Plus,
  Pointer,
  Popin,
  QuestionBold,
  Question,
  QuoteLeft,
  QuoteRight,
  RefreshBold,
  Refresh,
  Satchel,
  Schedule,
  SearchBold,
  Search,
  Shop,
  Sun,
  Support,
  Trees,
  Trophy,
  Truck,
  Tv,
  UserBold,
  Users,
  User,
  Wallet,
  Weight,
  Wheels,
} from './components';

export const ArrowBottomIcon = (props) => <IconWrapper {...props}><ArrowBottom /></IconWrapper>
export const ArrowCircleLeftIcon = (props) => <IconWrapper {...props}><ArrowCircleLeft /></IconWrapper>
export const ArrowCircleLeftBoldIcon = (props) => <IconWrapper {...props}><ArrowCircleLeftBold /></IconWrapper>
export const ArrowCircleRightBoldIcon = (props) => <IconWrapper {...props}><ArrowCircleRightBold /></IconWrapper>
export const ArrowCircleRightIcon = (props) => <IconWrapper {...props}><ArrowCircleRight /></IconWrapper>
export const ArrowLeftIcon = (props) => <IconWrapper {...props}><ArrowLeft /></IconWrapper>
export const ArrowLongBottomIcon = (props) => <IconWrapper {...props}><ArrowLongBottom /></IconWrapper>
export const ArrowLongLeftIcon = (props) => <IconWrapper {...props}><ArrowLongLeft /></IconWrapper>
export const ArrowLongRightIcon = (props) => <IconWrapper {...props}><ArrowLongRight /></IconWrapper>
export const ArrowLongTopIcon = (props) => <IconWrapper {...props}><ArrowLongTop /></IconWrapper>
export const ArrowRightIcon = (props) => <IconWrapper {...props}><ArrowRight /></IconWrapper>
export const ArrowRoundBottomLeftIcon = (props) => <IconWrapper {...props}><ArrowRoundBottomLeft /></IconWrapper>
export const ArrowRoundBottomRightIcon = (props) => <IconWrapper {...props}><ArrowRoundBottomRight /></IconWrapper>
export const ArrowRoundTopLeftIcon = (props) => <IconWrapper {...props}><ArrowRoundTopLeft /></IconWrapper>
export const ArrowRoundTopRightIcon = (props) => <IconWrapper {...props}><ArrowRoundTopRight /></IconWrapper>
export const ArrowTopIcon = (props) => <IconWrapper {...props}><ArrowTop /></IconWrapper>
export const BalloonBoldIcon = (props) => <IconWrapper {...props}><BalloonBold /></IconWrapper>
export const BalloonIcon = (props) => <IconWrapper {...props}><Balloon /></IconWrapper>
export const BankIcon = (props) => <IconWrapper {...props}><Bank /></IconWrapper>
export const BeakerBoldIcon = (props) => <IconWrapper {...props}><BeakerBold /></IconWrapper>
export const BeakerIcon = (props) => <IconWrapper {...props}><Beaker /></IconWrapper>
export const BellBoldIcon = (props) => <IconWrapper {...props}><BellBold /></IconWrapper>
export const BellIcon = (props) => <IconWrapper {...props}><Bell /></IconWrapper>
export const BulbBoldIcon = (props) => <IconWrapper {...props}><BulbBold /></IconWrapper>
export const BulbIcon = (props) => <IconWrapper {...props}><Bulb /></IconWrapper>
export const CardsBoldIcon = (props) => <IconWrapper {...props}><CardsBold /></IconWrapper>
export const CardsIcon = (props) => <IconWrapper {...props}><Cards /></IconWrapper>
export const CarIcon = (props) => <IconWrapper {...props}><Car /></IconWrapper>
export const CartIcon = (props) => <IconWrapper {...props}><Cart /></IconWrapper>
export const CautionBoldIcon = (props) => <IconWrapper {...props}><CautionBold /></IconWrapper>
export const CautionIcon = (props) => <IconWrapper {...props}><Caution /></IconWrapper>
export const ChangeIcon = (props) => <IconWrapper {...props}><Change /></IconWrapper>
export const CheckBoldIcon = (props) => <IconWrapper {...props}><CheckBold /></IconWrapper>
export const CheckboxIcon = (props) => <IconWrapper {...props}><Checkbox /></IconWrapper>
export const CheckIcon = (props) => <IconWrapper {...props}><Check /></IconWrapper>
export const ClockBoldIcon = (props) => <IconWrapper {...props}><ClockBold /></IconWrapper>
export const ClockIcon = (props) => <IconWrapper {...props}><Clock /></IconWrapper>
export const CodeIcon = (props) => <IconWrapper {...props}><Code /></IconWrapper>
export const CoinsIcon = (props) => <IconWrapper {...props}><Coins /></IconWrapper>
export const ContractIcon = (props) => <IconWrapper {...props}><Contract /></IconWrapper>
export const CrossBoldIcon = (props) => <IconWrapper {...props}><CrossBold /></IconWrapper>
export const CrossIcon = (props) => <IconWrapper {...props}><Cross /></IconWrapper>
export const DownloadIcon = (props) => <IconWrapper {...props}><Download /></IconWrapper>
export const EmptyIcon = (props) => <IconWrapper {...props}><Empty /></IconWrapper>
export const EyeIcon = (props) => <IconWrapper {...props}><Eye /></IconWrapper>
export const FactoryIcon = (props) => <IconWrapper {...props}><Factory /></IconWrapper>
export const GlobeIcon = (props) => <IconWrapper {...props}><Globe /></IconWrapper>
export const HeartIcon = (props) => <IconWrapper {...props}><Heart /></IconWrapper>
export const HourglassIcon = (props) => <IconWrapper {...props}><Hourglass /></IconWrapper>
export const InformationIcon = (props) => <IconWrapper {...props}><Information /></IconWrapper>
export const LeafIcon = (props) => <IconWrapper {...props}><Leaf /></IconWrapper>
export const LetterBoldIcon = (props) => <IconWrapper {...props}><LetterBold /></IconWrapper>
export const LetterIcon = (props) => <IconWrapper {...props}><Letter /></IconWrapper>
export const LinkBoldIcon = (props) => <IconWrapper {...props}><LinkBold /></IconWrapper>
export const LinkIcon = (props) => <IconWrapper {...props}><Link /></IconWrapper>
export const ListIcon = (props) => <IconWrapper {...props}><List /></IconWrapper>
export const LockIcon = (props) => <IconWrapper {...props}><Lock /></IconWrapper>
export const MarkerIcon = (props) => <IconWrapper {...props}><Marker /></IconWrapper>
export const MealIcon = (props) => <IconWrapper {...props}><Meal /></IconWrapper>
export const MegaphoneIcon = (props) => <IconWrapper {...props}><Megaphone /></IconWrapper>
export const MenuBoldIcon = (props) => <IconWrapper {...props}><MenuBold /></IconWrapper>
export const MenuDotsIcon = (props) => <IconWrapper {...props}><MenuDots /></IconWrapper>
export const MenuIcon = (props) => <IconWrapper {...props}><Menu /></IconWrapper>
export const MeterIcon = (props) => <IconWrapper {...props}><Meter /></IconWrapper>
export const MinusIcon = (props) => <IconWrapper {...props}><Minus /></IconWrapper>
export const MoonIcon = (props) => <IconWrapper {...props}><Moon /></IconWrapper>
export const MooveIcon = (props) => <IconWrapper {...props}><Moove /></IconWrapper>
export const OrganizationIcon = (props) => <IconWrapper {...props}><Organization /></IconWrapper>
export const OutIcon = (props) => <IconWrapper {...props}><Out /></IconWrapper>
export const PaperclipIcon = (props) => <IconWrapper {...props}><Paperclip /></IconWrapper>
export const PenIcon = (props) => <IconWrapper {...props}><Pen /></IconWrapper>
export const PhoneIcon = (props) => <IconWrapper {...props}><Phone /></IconWrapper>
export const PictureIcon = (props) => <IconWrapper {...props}><Picture /></IconWrapper>
export const PlaneIcon = (props) => <IconWrapper {...props}><Plane /></IconWrapper>
export const PlusIcon = (props) => <IconWrapper {...props}><Plus /></IconWrapper>
export const PointerIcon = (props) => <IconWrapper {...props}><Pointer /></IconWrapper>
export const PopinIcon = (props) => <IconWrapper {...props}><Popin /></IconWrapper>
export const QuestionBoldIcon = (props) => <IconWrapper {...props}><QuestionBold /></IconWrapper>
export const QuestionIcon = (props) => <IconWrapper {...props}><Question /></IconWrapper>
export const QuoteLeftIcon = (props) => <IconWrapper {...props}><QuoteLeft /></IconWrapper>
export const QuoteRightIcon = (props) => <IconWrapper {...props}><QuoteRight /></IconWrapper>
export const RefreshBoldIcon = (props) => <IconWrapper {...props}><RefreshBold /></IconWrapper>
export const RefreshIcon = (props) => <IconWrapper {...props}><Refresh /></IconWrapper>
export const SatchelIcon = (props) => <IconWrapper {...props}><Satchel /></IconWrapper>
export const ScheduleIcon = (props) => <IconWrapper {...props}><Schedule /></IconWrapper>
export const SearchBoldIcon = (props) => <IconWrapper {...props}><SearchBold /></IconWrapper>
export const SearchIcon = (props) => <IconWrapper {...props}><Search /></IconWrapper>
export const ShopIcon = (props) => <IconWrapper {...props}><Shop /></IconWrapper>
export const SunIcon = (props) => <IconWrapper {...props}><Sun /></IconWrapper>
export const SupportIcon = (props) => <IconWrapper {...props}><Support /></IconWrapper>
export const TreesIcon = (props) => <IconWrapper {...props}><Trees /></IconWrapper>
export const TrophyIcon = (props) => <IconWrapper {...props}><Trophy /></IconWrapper>
export const TruckIcon = (props) => <IconWrapper {...props}><Truck /></IconWrapper>
export const TvIcon = (props) => <IconWrapper {...props}><Tv /></IconWrapper>
export const UserBoldIcon = (props) => <IconWrapper {...props}><UserBold /></IconWrapper>
export const UsersIcon = (props) => <IconWrapper {...props}><Users /></IconWrapper>
export const UserIcon = (props) => <IconWrapper {...props}><User /></IconWrapper>
export const WalletIcon = (props) => <IconWrapper {...props}><Wallet /></IconWrapper>
export const WeightIcon = (props) => <IconWrapper {...props}><Weight /></IconWrapper>
export const WheelsIcon = (props) => <IconWrapper {...props}><Wheels /></IconWrapper>
