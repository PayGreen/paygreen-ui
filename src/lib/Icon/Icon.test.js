import React from 'react';
import TestRenderer from 'react-test-renderer';
import { ThemeDefault } from '../../theme';
import { ShopIcon } from './Icon';

it('renders without crashing', () => {
    const icon = TestRenderer.create(
        <ShopIcon
            theme={ThemeDefault}
        />
    );
    expect(icon.toJSON()).toMatchSnapshot();
});