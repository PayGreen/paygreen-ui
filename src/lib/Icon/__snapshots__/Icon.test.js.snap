// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<span
  className="style__IconBase-ak3679-0 hftxTk icon"
  type={null}
>
  <svg
    viewBox="0 0 143 143"
  >
    <path
      d="M123.2 55.5L110.8 25c-.5-1.3-1.8-2.2-3.2-2.2H33.2c-1.4 0-2.7.9-3.2 2.2L17.8 55.5c-.2.4-.3.9-.3 1.3 0 7.4 6 13.4 13.4 13.4 3.9 0 7.4-1.7 9.9-4.4 2.5 2.7 6 4.4 9.9 4.4 4 0 7.6-1.8 10-4.5 2.5 2.8 6 4.5 10 4.5 3.8 0 7.4-1.7 9.8-4.4 2.4 2.7 6 4.4 9.9 4.4s7.4-1.7 9.9-4.3c2.5 2.7 6 4.3 9.9 4.3 7.4 0 13.4-6 13.4-13.4-.1-.5-.2-.9-.4-1.3zm-13.1 7.7c-3.5 0-6.3-2.8-6.4-6.1 0-.4 0-.5-1.9-8.6-.8-3.3-1.7-7-1.8-7.5-.3-1.6-1.7-2.8-3.4-2.8-1.9 0-3.5 1.6-3.5 3.5 0 .4 0 .5 1.9 8.5.7 2.8 1.4 6.1 1.7 7.3-.3 3.3-3 5.8-6.3 5.8-3.5 0-6.3-2.7-6.4-6.1 0-.2 0-.3-.5-6.5-.2-2.2-.5-5.2-.8-9.3-.2-1.9-1.8-3.3-3.6-3.2-1.9.1-3.3 1.6-3.3 3.5v.7c.1.7.2 2.2.5 5.4.1.6.1 1.3.2 2 .3 3.6.6 6.7.6 7.5-.2 3.4-3 6.1-6.4 6.1-3.5 0-6.3-2.8-6.4-6.3L65.4 42c.1-1.9-1.3-3.6-3.2-3.7-1.9-.2-3.6 1.3-3.8 3.2l-1.4 15v.3c0 3.5-2.9 6.4-6.4 6.4-3.4 0-6.1-2.6-6.4-5.9.3-1.1 1-4.3 1.7-7.2.8-3.6 1.3-5.5 1.6-6.7.1-.4.2-.7.3-1 .1-.5.1-.6.1-.8 0-1.8-1.3-3.3-3.1-3.5s-3.4.9-3.8 2.7c-1.7 7.2-2.6 11.1-3.1 13.3-.6 2.6-.6 2.7-.6 3 0 3.5-2.9 6.3-6.4 6.3-3.3 0-6-2.5-6.3-5.8l11-27.5h69.7l11.2 27.5c-.3 3-3.1 5.6-6.4 5.6zM115 74.3c-1.9 0-3.5 1.6-3.5 3.5v36.4H52.7V77.6c0-1.9-1.6-3.5-3.5-3.5s-3.5 1.6-3.5 3.5v36.5H29.4V77.8c0-1.9-1.6-3.5-3.5-3.5s-3.5 1.6-3.5 3.5v39.9c0 1.9 1.6 3.5 3.5 3.5H115c1.9 0 3.5-1.6 3.5-3.5V77.8c0-2-1.6-3.5-3.5-3.5z"
    />
    <path
      d="M100.8 104.7c2.6 0 4.7-2.1 4.7-4.7V79.8c0-2.6-2.1-4.7-4.7-4.7H63.2c-2.6 0-4.7 2.1-4.7 4.7V100c0 2.6 2.1 4.7 4.7 4.7h37.6zM65.5 82.1h33.1v15.6H65.5V82.1z"
    />
  </svg>
</span>
`;
